### PR TITLE
Add module initializers tests

### PR DESCRIPTION
**Pull request by RikkiGibson**
Filling out test plan #40500

### CODE REVIEW

**Comment by RikkiGibson** [comment]
@dotnet/roslyn-compiler Could I please get a review of this test-only change?

File: src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializers/AccessibilityTests.cs [hunk 1]
```diff
@@ -122,6 +122,114 @@ namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : S
...
+    static partial void M2();
+    static partial void M2() { }
+
+    static partial void M3();
+    [ModuleInitializer] // 3
+    static partial void M3() { }
+
+    [ModuleInitializer] // 4
+    static partial void M4();
+    [ModuleInitializer] // 5
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializers/AccessibilityTests.cs:147]: >[ModuleInitializer] // 5 [](start = 4, length = 24)

Do we have a test for a scenario when the second application is allowed

File: src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializers/AccessibilityTests.cs [hunk 2]
```diff
@@ -122,6 +122,114 @@ namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : S
...
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+partial class C
+{
+    [ModuleInitializer]
+    public static partial void M1();
+    [ModuleInitializer] // 1
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializers/AccessibilityTests.cs:219]: >[ModuleInitializer] // 1 [](start = 4, length = 24)

Do we have a test for a scenario when the second application is allowed?

**Comment by RikkiGibson** [comment]
Thanks, will open a new PR to add the additional tests.

**Pull request merged**